tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -31,27 +46,34 @@ public class JsonDrivenTestFactory
         private readonly Dictionary<string, object> _objectMap;
         private readonly IJsonDrivenTestRunner _testRunner;
         private readonly EventCapturer _eventCapturer;
+        private readonly JsonDrivenTestsContext _testsContext;
 
         // public constructors
         public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap)
-            : this(null, client, databaseName, collectionName, bucketName, objectMap)
+            : this(client, databaseName, collectionName, bucketName, objectMap, null)
         {
         }
 
         public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap, EventCapturer eventCapturer)
-            : this(client, databaseName, collectionName, bucketName, objectMap)
+            : this(null, client, databaseName, collectionName, bucketName, objectMap, eventCapturer)
         {
-            _eventCapturer = eventCapturer;
         }
 
         public JsonDrivenTestFactory(IJsonDrivenTestRunner testRunner, IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap)
+            : this(testRunner, client, databaseName, collectionName, bucketName, objectMap, null)
+        {
+        }
+
+        public JsonDrivenTestFactory(IJsonDrivenTestRunner testRunner, IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap, EventCapturer eventCapturer)
```

Comment:
Too many constructors?

With this many constructors it is too easy for some client to call the wrong one.